src/argutil.c

```diff
@@ -67,6 +67,19 @@ int dcc_argv_search(char **a,
 }
 
 
+/**
+ * Return true if argv contains argument starting with needle.
+ */
+int dcc_argv_startswith(char **a,
+                        const char *needle)
+{
+    size_t needle_len = strlen(needle);
+    for (; *a; a++)
```

Comment:
This is unnecessary. `strncmp` does this for you.

Reply:
Let's check it:

```c
char *args[] = { "clang", "-c", "--target=aarch64-linux-gnu", "foo.c", NULL };
int idx = 2;
if (dcc_argv_startswith(args, "--target=") != 1) {
    fprintf(stderr, "wrong answer: args[%d] ('%s') does contain '--target'\n", idx, args[idx]);
    exit(1);
}
```

Removing `for` loop from `distcc_argv_startswith` breaks the test.
Perhaps I'm missing something. Could you elaborate, please? What exactly strncmp is supposed to do?

Reply:
oh my bad, I didn't see the double star.